jenkins/bootstrap.py

```diff
@@ -0,0 +1,514 @@
+
+    return 'unknown'
+
+
+class Paths(object):
+    """Links to remote gcs-paths for uploading results."""
+    def __init__(
+        self,
+        artifacts,  # artifacts folder (in build)
+        build_log,  # build-log.txt (in build)
+        build_path,  # path to build
+        finished,  # finished.json (metadata from end of build)
+        latest,   # latest-build.txt (in job)
+        pr_build_link,  # file containng build_path (in job directory)
+        pr_latest,  # latest-build.txt (in pr job)
+        pr_result_cache,  # jobResultsCache.json (in pr job)
+        result_cache,  # jobResultsCache.json (cache of latest results in job)
+        started,  # started.json  (metadata from start of build)
+    ):
+        self.artifacts = artifacts
+        self.build_log = build_log
+        self.build_path = build_path
+        self.finished = finished
+        self.latest = latest
+        self.pr_build_link = pr_build_link
+        self.pr_latest = pr_latest
+        self.pr_result_cache = pr_result_cache
+        self.result_cache = result_cache
+        self.started = started
+
+
+
+def CIPaths(job, build):
+    base = 'gs://kubernetes-jenkins/logs'
+    latest = os.path.join(base, job, 'latest-build.txt')
+    return Paths(
+        artifacts=os.path.join(base, job, build, 'artifacts'),
+        build_log=os.path.join(base, job, build, 'build-log.txt'),
+        build_path=None,
+        finished=os.path.join(base, job, build, 'finished.json'),
+        latest=latest,
+        pr_build_link=None,
+        pr_latest=None,
+        pr_result_cache=None,
+        result_cache=os.path.join(base, job, 'jobResultsCache.json'),
+        started=os.path.join(base, job, build, 'started.json'),
+    )
+
+
+
+def PRPaths(job, build, pull):
+    base = 'gs://kubernetes-jenkins/pr-logs'
+    build_path = os.path.join(base, 'pull', pull, job, build)
+    result_cache = os.path.join(
+            base, 'directory', job, 'jobResultsCache.json')
+    pr_result_cache = os.path.join(
+            base, 'pull', pull, job, 'jobResultsCache.json')
+    return Paths(
+        artifacts=os.path.join(build_path, 'artifacts'),
+        build_log=os.path.join(build_path, 'build-log.txt'),
+        build_path=build_path,
+        finished=os.path.join(build_path, 'finished.json'),
+        latest=os.path.join(base, 'directory', job, 'latest-build.txt'),
+        pr_build_link=os.path.join(base, 'directory', job, '%s.txt' % build),
+        pr_latest=os.path.join(base, 'pull', pull, job, 'latest-build.txt'),
+        pr_result_cache=pr_result_cache,
+        result_cache=result_cache,
+        started=os.path.join(build_path, 'started.json'),
+    )
+
+
+
+def Uniq():
+    """Return a probably unique suffix for the process."""
+    return '%x-%d' % (hash(Node()), os.getpid())
+
+
+BUILD_ENV = 'BUILD_NUMBER'
+BOOTSTRAP_ENV = 'BOOTSTRAP_MIGRATION'
+CLOUDSDK_ENV = 'CLOUDSDK_CONFIG'
+GCE_PRIVATE_KEY = 'JENKINS_GCE_SSH_PRIVATE_KEY_FILE'
```

Comment:
minor nit: this one and `SERVICE_ACCOUNT_PATH` should maybe have `_ENV` to be consistent with the other vars?